hazelcast/include/hazelcast/client/IMap.h

```diff
@@ -1034,7 +1033,7 @@ namespace hazelcast {
              * @see ICompletableFuture
              * @see #setAsync(Object, Object)
              */
-            std::shared_ptr<ICompletableFuture<V> > putAsync(const K &key, const V &value) {
+            future<std::shared_ptr<V>> putAsync(const K &key, const V &value) {
```

Comment:
Would it make sense to use shared_future here ?

Reply:
I try not to use shared_future where future is enough. the user can convert it to shared_future using future::share() method anyway.